### PR TITLE
Bump CAPV to v1.7.4

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*

*Description of changes:*
 Bumping CAPV to version v1.7.4
 
 Additionally, 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

- Ran `make build` successfully for the project.

### CODE REVIEW

File: projects/kubernetes-sigs/cluster-api-provider-vsphere/build/create_manifests.sh [hunk 1]
```diff
@@ -35,10 +35,14 @@ cd $REPO
 
 yq eval -i -P ".spec.template.spec.containers[0].args += [\"--namespace=eksa-system\"]" config/manager/manager.yaml
 
-make manifests STAGE="release" \
-  MANIFEST_DIR="out" \
-  PULL_POLICY="IfNotPresent" \
-  IMAGE="${IMAGE_REPO}/kubernetes-sigs/cluster-api-provider-vsphere/release/manager:$IMAGE_TAG"
+make manifest-modification STAGE="release" \
```
Comment (jaxesn) [inline projects/kubernetes-sigs/cluster-api-provider-vsphere/build/create_manifests.sh:38]: does these still build to and create files which are similiar to what we were producing? probably best to diff what this creates with what we shipped in 0.18.

File: projects/kubernetes-sigs/cluster-api-provider-vsphere/patches/0001-Add-capv-support-for-Bottlerocket.patch [hunk 2]
```diff
@@ -26,46 +24,6 @@ index 999f9846..09fe9c39 100644
  		}
  	}
  	if ctx.VSphereVM.Spec.CustomVMXKeys != nil {
-diff --git a/test/e2e/data/infrastructure-vsphere/capi-upgrades/v1alpha3/cluster-template.yaml b/test/e2e/data/infrastructure-vsphere/capi-upgrades/v1alpha3/cluster-template.yaml
```
[the remark this answers is not in the document]
Comment (cxbrowne1207) [reply projects/kubernetes-sigs/cluster-api-provider-vsphere/patches/0001-Add-capv-support-for-Bottlerocket.patch:29]: Yeah, it looks like it. This file was removed in [this](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2250) PR and included in v1.7.3 onwards

File: projects/kubernetes-sigs/cluster-api-provider-vsphere/build/create_manifests.sh (hunk 1 above)
Comment (cxbrowne1207) [reply projects/kubernetes-sigs/cluster-api-provider-vsphere/build/create_manifests.sh:38]: Yeah, they basically split this command and the combination of these two results in the same outputs from what I checked after running the `make build` command. I'll double check with a diff we shipped in 0.18

**Review by jaxesn — APPROVED** [review]

**Pull request merged**